java/arcs/core/allocator/Arc.kt

```diff
@@ -90,26 +92,23 @@ class Arc internal constructor(
     /** Called whenever the [ArcState] changes to [Error]. */
     suspend fun onError(handler: () -> Unit) = onArcStateChangeFiltered(Error, handler)
 
-    private fun fireArcStateChange() {
-        sync(this) {
-            arcStateChangeHandlers.forEach {
-                it(arcState)
-            }
-        }
+    private fun fireArcStateChange() = arcStateChangeHandlers.forEach {
```

Comment:
Is it safe to walk through the `List` w/o holding the lock?

Reply:
I refactored it to use a clone of the list. I don't really want to invoke the handlers (user code) from within the lock, since they could cause blocking behavior. This should prevent any iteration races on the list.